drafts/2.4-representable-functors.md

```diff
@@ -2,30 +2,34 @@
 type Reader a x = a -> x
 ```
 ```kotlin:ank:silent
-
+typealias Reader<D, A> = Kleisli<ForId, D, A>
 ```
 ................
 ```Haskell
 instance Functor (Reader a) where
     fmap f h = f . h
 ```
 ```kotlin:ank:silent
-
+fun <D, A> fmap(run: (D) -> A): Reader<D, A> = Kleisli(run.andThen { Id(it) })
 ```
 ................
 ```Haskell
 type Op a x = x -> a
 ```
 ```kotlin:ank:silent
-
+typealias Op<A, X> = (X) -> A
```

Comment:
I suggest using Op<A, X> as a data class, because we're using it in a Higherkinded position in `OpContravariant`. But your right to define it as a type alias.
Additionally, you may define the hkt version in the dedicated folder src/kotlin/higherkinds with the annotation, so your playground can resolve the import for `ForOp`.